### PR TITLE
Let prebuilds be cancelled

### LINKED ISSUE #19556: There is no cancel button on the new prebuild screen
### Bug description

When testing a prebuild that contained checking a condition to stop (the output of a program) I suddenly realized that that output state will never be reached. But there is no cancel button on the new screen. Luckily the old screen is still available. It would be very nice to have a cancel button on the new prebuild screen.

### Steps to reproduce

Create workspace with a prebuild.
Start the prebuild via the new screen
No way of stopping it.

### Workspace affected

_No response_

### Expected behavior

A cancel button in the prebuild screen of a specific prebuild.

### Example repository

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

Woodoo magic to make prebuild users happy.

## Related Issue(s)

Fixes EXP-1633
Fixes #19556

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-cancel-prebuilds</li>
	<li><b>🔗 URL</b> - <a href="https://ft-cancel-prebuilds.preview.gitpod-dev.com/workspaces" target="_blank">ft-cancel-prebuilds.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-cancel-prebuilds-gha.23783</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-cancel-prebuilds%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/prebuilds/detail/PrebuildDetailPage.tsx [hunk 1]
```diff
@@ -118,6 +124,18 @@ export const PrebuildDetailPage: FC = () => {
         }
     }, [isTriggerError, triggerError, toast]);
 
+    const cancelPrebuild = useCallback(async () => {
```
Comment (mustard-mh) [inline components/dashboard/src/prebuilds/detail/PrebuildDetailPage.tsx:127]: We can make `!prebuild` check inside useMutation, and remove this callback

**Comment by mustard-mh** [comment]
When I click cancel
- logs keep logging for seconds.
- logs loading indicator is in loading state forever
image

**Comment by mustard-mh** [comment]
Not sure how things going on server and ws-manager. But it's confused that I (as a user) am not sure if workspace still running, will my credit wasted?

**Comment by mustard-mh** [comment]
> To resolve it, we may need to change [code here](https://github.com/gitpod-io/gitpod/blob/ft/cancel-prebuilds/components/ws-manager-bridge/src/prebuild-state-mapper.ts#L43-L50) -> if pod only after pod is actully killed, we update it to final state

Not sure if it's worth doing that update (I'm not sure what will be affected), or if we have way to tell user this can happen, and it takes seconds (around 15s) to actual stop the prebuild

**Comment by filiptronicek** [comment]
> Not sure if it's worth doing that update (I'm not sure what will be affected), or if we have way to tell user this can happen, and it takes seconds (around 15s) to actual stop the prebuild

@mustard-mh do you think in this case we should stop streaming and optimistically lie to users before actually terminating?

**Comment by mustard-mh** [comment]
> do you think in this case we should stop streaming and optimistically lie to users before actually terminating?

Maybe, to avoid confusing

**Review by mustard-mh — APPROVED** [review]
LGTM

**Comment by filiptronicek** [comment]
I hope to get this merged soon, blocked by other work 🙏

**Pull request merged**